test/msgs.test.ts

```diff
@@ -73,4 +143,201 @@ describe('TLV', () => {
       expect(tlvFromExpected.value).toEqual(tlv.value);
     }
   });
+
+  test('readFromSocket', async () => {
+    const sendTLV = async (bytes: Uint8Array): Promise<void> => {
+      const ip = '127.0.0.1';
+      const port = 4000;
+
+      const server = TCP.createServer((sock: TCP.Socket) => {
+        sock.write(bytes);
+      });
+      server.listen(port, ip);
+      const sockClient = new TCP.Socket();
+      return new Promise((resolve, reject) => {
+        sockClient.connect(port, ip, () => {});
+        sockClient.on('readable', () => {
+          try {
+            TLV.readFromSocket(sockClient);
```

Comment:
I'd add an expect here that the value read here matches what we expect based on `bytes` (i.e. using `deserialize`). for the correct format test case